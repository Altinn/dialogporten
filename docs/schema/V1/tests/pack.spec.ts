import { expect, test } from 'vitest';
import { promisify } from 'util';
import { exec } from 'child_process';
const execP = promisify(exec);

test('Snapshot test of `npm pack --dry-run --json`', async () => {
    const result = await execP('npm pack --dry-run --json');
    const json = JSON.parse(result.stdout);

    // Expect no errors while packing
    expect(result.stderr).toBe("");

    // Expect there to be one pack
    expect(json.length).toBe(1);

    // Expect the list of files to be the same as last time
    //  - Testing filename and file mode, while ignoring file size
    const files = json[0].files.map(({ path, mode }) => ({ path, mode }));
    expect(files).toMatchSnapshot();
});
