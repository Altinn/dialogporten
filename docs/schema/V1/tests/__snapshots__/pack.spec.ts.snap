// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test of \`npm pack --dry-run --json\` 1`] = `
[
  {
    "mode": 420,
    "path": "README.md",
  },
  {
    "mode": 420,
    "path": "package.json",
  },
  {
    "mode": 420,
    "path": "schema.verified.graphql",
  },
  {
    "mode": 420,
    "path": "src/index.js",
  },
  {
    "mode": 420,
    "path": "swagger.verified.json",
  },
]
`;
